xbmc/network/EventServer.h

```diff
@@ -57,20 +59,19 @@ namespace EVENTSERVER
     int GetNumberOfClients();
 
   protected:
-    CEventServer();
     void Cleanup();
     void Run();
     void ProcessPacket(SOCKETS::CAddress& addr, int packetSize);
     void ProcessEvents();
     void RefreshClients();
 
-    std::map<unsigned long, EVENTCLIENT::CEventClient*>  m_clients;
-    static CEventServer* m_pInstance;
-    SOCKETS::CUDPSocket* m_pSocket;
+    std::map<unsigned long, std::unique_ptr<EVENTCLIENT::CEventClient>> m_clients;
+    static std::unique_ptr<CEventServer> m_pInstance;
```

Comment:
Can we move this to a container like servicemanager? As a side note, static members usually shouldn't be mixed with non-static members due to the different lifetimes.